File: LeetCode-Java/Jump Game.java
public class Solution {
    public boolean canJump(int[] A) {
        int currentMax = 0;
        for(int i = 0;i < A.length;i ++)
        {
            if(i > currentMax) return false;
            currentMax = Math.max(currentMax, i + A[i]);
            if(currentMax >= A.length - 1) return true;
        }
        return false;
    }
}

File: Leetcode-cpp/jump_game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int global_max = 0, start = 0;
        while (start <= global_max && global_max < nums.size() - 1) {
            global_max = max(global_max, start + nums[start]);
            start++;
        }
        return global_max >= nums.size() - 1;
    }
};